### DISCUSSION TITLE
Thinking about reasons that Orchard team use Vue.js over React.js.

### DISCUSSION THREAD

**Discussion by aaronamm** [OrchardCMS/OrchardCore#15917] [General]
Hello everyone.

I have tried to find reasons why Orchard team use Vue.js and do not use React.js which the leading in JS frontend eco system and plenty of plugins. Here is an example from [npmtrends.com](https://npmtrends.com/react-vs-vue)

Please correct me if I am wrong. 
- Vue has [Stand alone script option](https://vuejs.org/guide/extras/ways-of-using-vue#standalone-script) that can plug to a framework like OC (Orchard Core) which use most of HTML page easily.  We only attach [Vue directive](https://github.com/vuejs/petite-vue?tab=readme-ov-file#vue-compatible) to existing HTML elements. 
We can't do this with React because all HTML elements must be in React component and it requires build process to get output script.
- We can inject a default value from server-side easily by setting Vue's data function. Here is an example that we use in [tags-editor.js](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Taxonomies/Assets/js/tags-editor.js#L12)
- We don't need to setup a build process for a simple custom script, deal with many dependencies in Node ecosystem. Then we can focus more on improving/adding more features to OC.

**Comment by aaronamm** [comment]
CC @Skrypt

**Reply by Skrypt** [reply]
Simply because we decided that we prefered Vue back then. There is absolutely no dependency on a JS framework on the global layout of the admin UI which allows you to use any framework you prefer to build your admin UI pages in your custom modules.

**Reply by Skrypt** [reply]
And I believe we use React for GraphiQL as an example.

**Reply by aaronamm** [reply]
@Skrypt thank you so much for the information. I appreciate that.
